src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -105,14 +110,38 @@ func (r *ClusterReconciler) Reconcile(
 		return ctrl.Result{}, err
 	}
 
-	observedNodes := make([]string, 0, len(observedPods.Items))
+	var node corev1.Node
+
+	observedNodesInternal := make([]string, 0, len(observedPods.Items))
+	observedNodesExternal := make([]string, 0, len(observedPods.Items))
 	// nolint:gocritic // the copies are necessary for further redpandacluster updates
 	for _, item := range observedPods.Items {
-		observedNodes = append(observedNodes, fmt.Sprintf("%s.%s", item.Name, svc.HeadlessServiceFQDN()))
+		observedNodesInternal = append(observedNodesInternal,
+			fmt.Sprintf("%s.%s", item.Name, headlessSvc.HeadlessServiceFQDN()))
+
+		err = r.Get(ctx, types.NamespacedName{Name: item.Spec.NodeName}, &node)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("unable to featch node info: %w", err)
+		}
+
+		externalIP, _ := resources.GetNodeIPs(&node)
```

Comment:
I think we should do this only if external connectivity is enabled?

Reply:
should we handle the case where this is empty? You just return empty string here so we would end up returning just ports? e.g. my local kind cluster (obviously) does not have the external ip but I might still enable the feature. I wonder what is good way to communicate this to the user 🤔

Reply:
I guard this chunk of external nodes with the:
```
if redpandaCluster.Spec.ExternalConnectivity && nodePortSvc.LastObservedState != nil {
```

To answer the second question. Should I create log in Reconcile func? Should I emit an event?

Reply:
Ping.

Reply:
hmm ... yeah, I think log perhaps? Because otherwise we'll be emitting the even in every call of the reconcile.